### PR TITLE
What's New for 'dotnet' docs in June

### PR DESCRIPTION
**Pull request by MaximRouiller**

### CODE REVIEW

**Comment by pkulikov** [comment]
The PR title should mention June, not May

File: docs/whats-new/toc.yml [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-      - name: December
-        href: 2019-12.md
+name: What's new
+href: index.yml
+expanded: true
+items:
+- name: 2020
+  items:
+  - name: June
+    href: dotnet-2020-06.md
```
Comment (pkulikov) [inline docs/whats-new/toc.yml:8]: (Another *.yml* file is correct)
```suggestion
    href: 2020-06.md
```

File: docs/whats-new/index.yml [hunk 2]
```diff
@@ -1,75 +1,74 @@
...
+    - text: April 2020
+      url: dotnet-2020-04.md
+    - text: March 2020
+      url: dotnet-2020-03.md
+    - text: February 2020
+      url: dotnet-2020-02.md
+    - text: January 2020
+      url: dotnet-2020-01.md
+    - text: December 2019
+      url: 2019-12.md
```
Comment (pkulikov) [inline docs/whats-new/index.yml:28]: The latest six months?
```suggestion
```

File: docs/whats-new/toc.yml [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
-      - name: March
-        href: dotnet-2020-03.md
-      - name: February
-        href: dotnet-2020-02.md
-      - name: January
-        href: dotnet-2020-01.md
-    - name: 2019
-      items:
-      - name: December
-        href: 2019-12.md
```
Comment (pkulikov) [inline docs/whats-new/toc.yml:20]: Also, should the PR actually **remove** the *2019-12.md* file?

Comment (MaximRouiller) [reply docs/whats-new/toc.yml:20]: It does not at the moment. It's a decision we have to make in the future if we want to keep the links active or not.

File: docs/whats-new/2020-06.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+- [kthy](https://github.com/kthy) - Kristian Thy (1)
+- [MahmodAbdou](https://github.com/MahmodAbdou) (1)
+- [Marusyk](https://github.com/Marusyk) - Roman Marusyk (1)
+- [MirkoCovizzi](https://github.com/MirkoCovizzi) - Mirko Covizzi (1)
+- [Niharikadutta](https://github.com/Niharikadutta) - Niharika Dutta (1)
+- [nlehnert1](https://github.com/nlehnert1) - Chloe Lehnert (1)
+- [nnpcYvIVl](https://github.com/nnpcYvIVl) (1)
+- [oboukli](https://github.com/oboukli) - Omar Boukli-Hacene (1)
+- [OzBob](https://github.com/OzBob) (1)
+- [robvet](https://github.com/robvet) - Rob Vettor (1)
```
Comment (scottaddie) [inline docs/whats-new/2020-06.md:107]:
```suggestion
```

**Review by scottaddie — CHANGES_REQUESTED** [review]

**Review by scottaddie — APPROVED** [review]

File: docs/whats-new/index.yml [hunk 5]
```diff
@@ -1,75 +1,72 @@
...
-          - text: Visual Studio Code release notes
-            url: https://code.visualstudio.com/updates
+### YamlMime:Landing
+
+title: .NET documentation - what's new?
+summary: Welcome to what's new in .NET docs. Use this page to quickly find the latest changes.
+metadata:
+  title: .NET documentation what's new?
+  description: Learn about new and updated content in .NET docs.
+  ms.date: 2020-04-22T00:00:00.0000000
```
Comment (Youssef1313) [inline docs/whats-new/index.yml:8]: Looks like the date is again in the ISO format instead of yyyy/MM/dd

File: docs/whats-new/index.yml [hunk 6]
```diff
@@ -1,75 +1,72 @@
...
+  - linkListType: concept
+    links:
+    - text: Microsoft docs contributor guide
+      url: https://docs.microsoft.com/contribute
+    - text: .NET docs contributor guide
+      url: https://docs.microsoft.com/contribute/dotnet/dotnet-contribute
+    - text: .NET API reference docs contributor guide
+      url: https://github.com/dotnet/dotnet-api-docs/blob/master/CONTRIBUTING.md
+    - text: .NET samples contributor guide
+      url: https://github.com/dotnet/samples/blob/master/CONTRIBUTING.md
```
Comment (Youssef1313) [inline docs/whats-new/index.yml:46]: This is unrelated to the PR.
But you may consider removing/replacing these links.
Both links redirects the reader to the one in this repo.
I'd only include the one in this repo.

File: docs/whats-new/toc.yml (hunk 3 above)
Comment (Youssef1313) [reply docs/whats-new/toc.yml:20]: > It does not at the moment. It's a decision we have to make in the future if we want to keep the links active or not.

It's actually deleted currently.

File: docs/whats-new/2020-06.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+- [Implement a DisposeAsync method](/dotnet/standard/garbage-collection/implementing-disposeasync) - Implement the async dispose pattern
+
+## Community contributors
+
+The following people contributed to the .NET docs in June 2020. Thank you! Learn how to contribute by following the links under "Get involved" in the [what's new landing page](index.yml).
+
+- [NextTurn](https://github.com/NextTurn) - Next Turn (27)
+- [Youssef1313](https://github.com/Youssef1313) - Youssef Victor (10)
+- [pkulikov](https://github.com/pkulikov) - Petr Kulikov (4)
+- [damabe](https://github.com/damabe) - Dan Mabee (2)
```
Comment (BillWagner) [inline docs/whats-new/2020-06.md:88]: Dan is a vendor:

```suggestion
```

File: docs/whats-new/toc.yml (hunk 3 above)
Comment (BillWagner) [reply docs/whats-new/toc.yml:20]: We should remove the files when they are removed from the TOC. It's a known issue that it will 404, but the traffic after 6 months is really low.

**Review by BillWagner — CHANGES_REQUESTED** [review]
This is great work @MaximRouiller I am impressed with the automation.

I found one vendor that should be removed from the list of community contributors. Also, we should remove the Dec 2019 md file.

Then, this is ready to merge.

**Review by BillWagner — APPROVED** [review]
